lib/index.js

```diff
@@ -28,11 +28,83 @@ function getForecast(city = null){
 }
 
 function displayForecast(data){
-  // debugger
-  // User Story / Issue #12
+
+  clearValue( byId( 'navSearch' ) )
+
+
+  let info = data['data']['attributes']
+
+  let location = info['location']
+  let today    = info['today']
+  let current  = info['current']
+  let forecast = info['forecast']['data']['attributes']
+
+  let showToday = {
+    'gif':     null,
+    'desc':    current['summary'],
+    'temp':    current['temperature'],
+    'high':    today['high'],
+    'low':     today['low'],
+    'city':    location['city']['long_name'],
+    'state':   location['state']['short_name'],
+    'country': location['country']['long_name'],
+    'time':    current['time'],
+  }
+
+  let content = byId('content')
+
+  let overview = addDiv(content, 'overview', null )
+
+  displayToday(showToday, overview)
+
+
 
 }
 
+  function displayToday(data, element) {
+
+    let cityState = data['city'] + ', ' + data['state']
+    let highLow = "High: " + data['high'] + " Low: " + data['low']
+
+    let showToday = addDiv(element, 'today', 'overview')
+
+    let today1 = addSpan(showToday, 'today1_temp', 'today')
+      // let today1_1 = addSpan(today1, 'today1_1', 'today1')
+      let today1_1 = addDiv(today1, 'today1_1', 'today1')
+        let today1_gif  = addSpan(today1_1, 'today1_gif', 'today1_1')
+        appendText(today1_gif, "IMG")// TO DO - add gif image
+        let today1_desc = addSpan(today1_1, 'today1_desc', 'today1_1')
+        appendText(today1_desc, data['desc'])
+      // let today1_2 = addSpan(today1, 'today1_2', 'today1')
+      let today1_2 = addDiv(today1, 'today1_2', 'today1')
+        appendText(today1_2, data['temp'])
+      // let today1_3 = addSpan(today1, 'today1_3', 'today1')
+      let today1_3 = addDiv(today1, 'today1_3', 'today1')
+        appendText(today1_3, highLow)
+
+    let today2 = addSpan(showToday, 'today2_location', 'today')
+      // let today2_1 = addSpan(today2, 'today2_1', 'today2')
+      let today2_1 = addDiv(today2, 'today2_1', 'today2')
+      appendText(today2_1, cityState)
+      // let today2_2 = addSpan(today2, 'today2_2', 'today2')
+      let today2_2 = addDiv(today2, 'today2_2', 'today2')
+      appendText(today2_2, data['country'])
+      // let today2_3 = addSpan(today2, 'today2_3', 'today2')
+      let today2_3 = addDiv(today2, 'today2_3', 'today2')
+      appendText(today2_3, data['time'])
+
+    let today3 = addSpan(showToday, 'today2_location', 'today')
+      // let today3_1 = addSpan(today3, 'today3_1', 'today3')
+      let today3_1 = addDiv(today3, 'today3_1', 'today3')
+      appendText(today3_1, "Change Location")
+      let today3_2 = addDiv(today3, 'today3_2', 'today3')
+      appendText(today3_2, "Add to Favorites")
+
```

Comment:
@corywest Hi Cory, can you review this?

First off, I hate what I'm doing here, this is a massive method and goes against everything I've learned so far, but I couldn't get require/import to work (like most of the class) and I didn't want to make methods for each of these pieces because naming and file structure started to feel cumbersome. 

Essentially, what's happening here is, I'm filling an empty div on my page with content for one of the Forecast boxes from the wire frame. I'm going to use flexbox to handle the positioning, which is why I have so many nested Divs/Spans. --- If there's a better way to do that, I haven't received feedback on that thus far and would love to hear.

Otherwise, my big questions with this method is - should I have a method that builds all these empty divs/spans before I try to fill them with another method? 

My concern is that this method gets called with from the API request, and thus only happens when the API returns. I guess, in this case, all of the content on that page (the other two forecast boxes) come from the same API, so I don't have to worry about the order of API's returning... 

But hypothetically, if I did need multiple API requests, would this HTML structure not necessarily end up being built in the section of the page I intend it for, because whichever API returned first would build first (ie append to container first)?

Reply:
![20190124_224819](https://user-images.githubusercontent.com/36523304/51727694-6531f980-202a-11e9-90d1-6ed33cb67a6d.jpg)

These are my flexbox & grid intentions for the two Today-related forecast boxes of the wireframe.

Reply:
This is deployed [here](https://kate-v2.github.io/sweater_weather_web/) which should give you access to my whole index.js file if that's easier to understand than this little snip